src/command_func.h

```diff
@@ -78,6 +78,19 @@ struct RecursiveCommandCounter {
 	static int _counter;
 };
 
+#if defined(__GNUC__) && !defined(__clang__)
+/*
```

Comment:
Might it be worth defining something here like `#define SILENCE_GCC_CLANG_FUNCTION_POINTER_CAST` or something, which then gets used in all other places where we push or pop the `pragma GCC diagonostics`. If at some point some other compiler needs it, you only need to change it in one location.